pkg/scheduler/scheduler.go

```diff
@@ -780,11 +783,23 @@ type podConditionUpdaterImpl struct {
 
 func (p *podConditionUpdaterImpl) update(pod *v1.Pod, condition *v1.PodCondition) error {
 	klog.V(3).Infof("Updating pod condition for %s/%s to (%s==%s, Reason=%s)", pod.Namespace, pod.Name, condition.Type, condition.Status, condition.Reason)
-	if podutil.UpdatePodCondition(&pod.Status, condition) {
-		_, err := p.Client.CoreV1().Pods(pod.Namespace).UpdateStatus(context.TODO(), pod, metav1.UpdateOptions{})
+	oldData, err := json.Marshal(pod)
+	if err != nil {
 		return err
 	}
-	return nil
+	if !podutil.UpdatePodCondition(&pod.Status, condition) {
+		return nil
+	}
+	newData, err := json.Marshal(pod)
+	if err != nil {
+		return err
+	}
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, &v1.Pod{})
+	if err != nil {
+		return fmt.Errorf("failed to CreateTwoWayMergePatch for pod %q/%q: %v", pod.Namespace, pod.Name, err)
+	}
+	_, err = p.Client.CoreV1().Pods(pod.Namespace).Patch(context.TODO(), pod.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{}, "status")
```

Comment:
what cases would fail that PATCH? I assume only when the version on the server had a different pod condition than the base version we used for the patch, right? meaning if pod.Spec.NodeName was modified, this would still succeed, correct?

Reply:
> if pod.Spec.NodeName was modified, this would still succeed, correct?

Yes. However, this Patch is called only by `recordSchedulingFailure()`, so I don't think .spec.nodeName will be set elsewhere.

Reply:
Are we saying that if this fails (there was another change), we still don't cancel the cycle?

I'm referring to https://github.com/kubernetes/kubernetes/blob/a15168288792daa23c993e551894229161dabdd0/pkg/scheduler/scheduler.go#L452-L456

Reply:
The question above hasn't been resolved yet.

Reply:
Using patch, I'm not sure what would cause an error in this case other than some kind of network/infrastructure error. 

A conflict won't occur anymore if something else in the Status has been changed, so the request should no longer ever fail with a 409 status code. 

I'll defer to those who have more experience with the scheduler, but in my opinion if an error here was ignored/logged before, we can just keep ignoring/logging the error.

Besides, if you wanted to handle this error and cancel the cycle at this point, it would mean undoing everything that has occurred up to that point in the preemption, which I'm not sure would even be reliably possible.

Reply:
> Are we saying that if this fails (there was another change), we still don't cancel the cycle?

Yes. It's like the operation is half-completed - preemptor get its NominatedNodeName set, but the pods that has lower priority and also had NominatedNodeName failed on clearing their NominatedNodeName field. Which is the root cause of this flake.